java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,431 @@
+                        break;
+                    case Types.CLOB:
+                        updateVector((VarCharVector)root.getVector(columnName),
+                            rs.getClob(i), rowCount);
+                        break;
+                    case Types.BLOB:
+                        updateVector((VarBinaryVector)root.getVector(columnName),
+                            rs.getBlob(i), rowCount);
+                        break;
+
+                    default:
+                        // no-op, shouldn't get here
+                        break;
+                }
+            }
+            rowCount++;
+        }
+        root.setRowCount(rowCount);
+    }
+
+    private static void updateVector(BitVector bitVector, boolean value, int rowCount) {
+        bitVector.setSafe(rowCount, value? 1: 0);
+        bitVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(TinyIntVector tinyIntVector, int value, int rowCount) {
+        tinyIntVector.setSafe(rowCount, value);
+        tinyIntVector.setValueCount(rowCount + 1);
+    }
+
+    private static  void updateVector(SmallIntVector smallIntVector, int value, int rowCount) {
+        smallIntVector.setSafe(rowCount, value);
+        smallIntVector.setValueCount(rowCount + 1);
+    }
+
+    private static  void updateVector(IntVector intVector, int value, int rowCount) {
+        intVector.setSafe(rowCount, value);
+        intVector.setValueCount(rowCount + 1);
+    }
+
+    private static  void updateVector(BigIntVector bigIntVector, int value, int rowCount) {
+        bigIntVector.setSafe(rowCount, value);
+        bigIntVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(DecimalVector decimalVector, BigDecimal value, int rowCount) {
+        decimalVector.setSafe(rowCount, value);
+        decimalVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(Float4Vector float4Vector, float value, int rowCount) {
+        float4Vector.setSafe(rowCount, value);
+        float4Vector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(Float8Vector float8Vector, double value, int rowCount) {
+        float8Vector.setSafe(rowCount, value);
+        float8Vector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(VarCharVector varcharVector, String value, int rowCount) {
+        if (value != null) {
+            varcharVector.setIndexDefined(rowCount);
+            varcharVector.setValueLengthSafe(rowCount, value.length());
+            varcharVector.setSafe(rowCount, value.getBytes(StandardCharsets.UTF_8), 0, value.length());
+            varcharVector.setValueCount(rowCount + 1);
+        }
+        // TODO: not sure how to handle null string value ???
+    }
+
+    private static void updateVector(DateMilliVector dateMilliVector, Date date, int rowCount) {
+        //TODO: Need to handle Timezone
```

Comment:
I think for java.sql.Date, it is safe to assume you can ignore timezone. According to jdbc spec/google searches, a JDBC driver would put the date as a Local date. Same for java.sql.Time...